hudi-hadoop-common/src/main/java/org/apache/hudi/hadoop/fs/HadoopFSUtils.java

```diff
@@ -171,6 +171,17 @@ public static StoragePathInfo convertToStoragePathInfo(FileStatus fileStatus) {
         fileStatus.getModificationTime());
   }
 
+  public static StoragePathInfo convertToStoragePathInfo(FileStatus fileStatus, String[] locations) {
```

Comment:
`LocatedFileStatus` (which extends `FileStatus`) stores `BlockLocation[] locations`.  As a follow-up, see if we want to leverage that.

Reply:
HUDI-8091